app/src/main/java/org/mozilla/fenix/perf/PerformanceInflater.kt

```diff
@@ -0,0 +1,68 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.perf
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import org.mozilla.fenix.ext.getAndIncrementNoOverflow
+import java.util.concurrent.atomic.AtomicInteger
+
+
+/**
+ * Counts the number of inflations fenix does. This class behaves only as an inflation counter since
+ * it takes the `inflater` that is given by the base system. This is done in order not to change
+ * the behavior of the app since all we want to do is count the inflations done.
+ *
+ */
+open class PerformanceInflater(val inflater: LayoutInflater, context: Context) : LayoutInflater(
+    inflater,
+    context
+){
+
+    private val sClassPrefixList = arrayOf(
+        "android.widget.",
+        "android.webkit.",
+        "android.app."
+    )
+
+    override fun cloneInContext(newContext: Context?): LayoutInflater {
+        return PerformanceInflater(inflater, newContext!!)
```

Comment:
So this is an interesting one, doing the `this` instead of the `inflater` leads to a multiple of issue because the phone passes us the `PhoneLayoutInflater`. By doing `this` (which the `PhoneLayoutInflater` does), we end up crashing by missing some classes in the `classLoader` (which is what crashes). I'm assuming because the inflater doesn't implement *everything* that needs to be done by an inflater

Reply:
This was right (again). It used to crash because of something else and I think this was just a symptom when I was investigating. I changed it to this and we don't lose any functionality now.